### PR TITLE
feat(docutils): add "full" watch mode

### PR DESCRIPTION
**Pull request by boneskull**
BREAKING CHANGE: This package is now ESM-only.

TypeDoc itself is unable to watch and rebuild a "composite" TS project (I'm not sure why, but the sources say something about "too hard").  Maybe just _too hard to do the right way_.

So I did it manually; gather a list of files to watch and re-run TypeDoc when needed.  Once TypeDoc outputs files, `mkdocs serve` will see the output and rebuild/reload itself accordingly.

Unfortunately--and this is a questionable decision on my part--the file watcher I found looks a lot nicer than the usual `chokidar`--but it is ESM only.  To use it, `@appium/docutils` must also be ESM-only (or rather, I haven't figured out how I can `await import(someEsmOnlyPkg)` in TS; it definitely depends on how a package distributes itself).  I expect eventually the ecosystem will move more towards ESM modules (for better or worse), and we can think of this as an experimental foray into that sort of thing.

Because `@appium/docutils` is mostly intended to be used as a CLI app, I think we can get away with this.  In `appium`'s `docs/scripts/build-docs.js`, we _do_ need to pull in `@appium/docutils` as a library, but that seems to work fine with `await import()`.

Tests don't run, so the build will break.  If we want to go down this rabbit hole further, I'll see
about getting them to work.  All-in-all it wasn't _too_ bad getting TS to do the right thing, but
maybe just using `chokidar` is for the best (though it pulls in native module(s)).

### CODE REVIEW

**Comment by boneskull** [comment]
Current dependencies on/for this PR:
* master
  * **PR #18432** <a href="https://app.graphite.dev/github/pr/appium/appium/18432" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/appium/appium/18432?utm_source=stack-comment).

File: packages/opencv/lib/index.js
Comment (boneskull) [inline packages/opencv/lib/index.js:1]: I need to pull this change out into its own PR.

**Comment by jlipps** [comment]
I feel a little hesitant about introducing ESM stuff without a clear strategy for how other tools, particularly in our own ecosystem, could import it. Is there a way for us to make this project ESM but ensure that somehow it also exports a CJS version?
